### PR TITLE
Add SCOPE_DSN support

### PR DESCRIPTION
**Pull request by tonyredondo**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -125,7 +126,15 @@ func NewAgent(options ...Option) (*Agent, error) {
...
 	if agent.apiKey == "" {
-		if apikey, set := os.LookupEnv("SCOPE_APIKEY"); set && apikey != "" {
+		if dsn, set := os.LookupEnv("SCOPE_DSN"); set && dsn != "" {
+			dsnApiKey, dsnApiEndpoint, dsnErr := parseDSN(dsn)
+			if dsnErr != nil {
+				agent.logger.Printf("Error parsing dsn value: %v\n", dsnErr)
+			} else {
+				agent.apiKey = dsnApiKey
+				agent.apiEndpoint = dsnApiEndpoint
+			}
```
Comment (fermayo) [inline agent/agent.go]: Can we move this out before `if agent.apiKey == "" {`, so we only do it once?

File: agent/agent.go [hunk 2]
```diff
@@ -125,7 +126,15 @@ func NewAgent(options ...Option) (*Agent, error) {
 	configProfile := GetConfigCurrentProfile()
 
 	if agent.apiKey == "" {
-		if apikey, set := os.LookupEnv("SCOPE_APIKEY"); set && apikey != "" {
+		if dsn, set := os.LookupEnv("SCOPE_DSN"); set && dsn != "" {
+			dsnApiKey, dsnApiEndpoint, dsnErr := parseDSN(dsn)
+			if dsnErr != nil {
+				agent.logger.Printf("Error parsing dsn value: %v\n", dsnErr)
```
Comment (fermayo) [inline agent/agent.go]:
```suggestion
				agent.logger.Printf("Error parsing dsn value: %v", dsnErr)
```

`log.Printf` does not need `\n`

File: agent/agent.go [hunk 3]
```diff
@@ -273,3 +290,16 @@ func generateAgentID() string {
 	}
 	return agentId.String()
 }
+
+func parseDSN(dsnString string) (apiKey string, apiEndpoint string, err error) {
```
Comment (fermayo) [inline agent/agent.go]: A unit test for this function would be nice

File: agent/agent.go (hunk 1 above)
Comment (tonyredondo) [reply agent/agent.go]: ok, in the next commit.

File: agent/agent.go (hunk 3 above)
Comment (tonyredondo) [reply agent/agent.go]: ok

File: agent/agent.go (hunk 1 above)
Comment (tonyredondo) [reply agent/agent.go]: done

File: agent/agent.go (hunk 3 above)
Comment (tonyredondo) [reply agent/agent.go]: done

**Pull request merged**